lib/ansible/playbook/role/definition.py

```diff
@@ -198,7 +198,8 @@ def _load_role_path(self, role_name):
             role_name = os.path.basename(role_name)
             return (role_name, role_path)
 
-        raise AnsibleError("the role '%s' was not found in %s" % (role_name, ":".join(role_search_paths)), obj=self._ds)
+        searches = self._collection_list + role_search_paths
```

Comment:
This will go bang if `self._collection_list` is None, which I think we force it not to be, but not 100% sure off the top of my head

Reply:
+1 I can take care of that